src/plugins/segmentio/index.ts

```diff
@@ -63,6 +65,32 @@ export function segmentio(
       ? batch(apiHost, settings?.deliveryStrategy?.config)
       : standard()
 
+  async function flushQueue(xt: Plugin, queue: PriorityQueue<Context>) {
```

Comment:
Make sure to test this behavior with batching enabled.
I'm curious to see what happens if you push 1000 events into the queue while offline